tools/disasm/sym_info.py

```diff
@@ -0,0 +1,217 @@
+            sym_vrom_int = int(sym_vrom, 0)
+
+            syms_by_section_by_file.setdefault(sym_file, dict()).setdefault(
+                sym_section, list()
+            ).append(
+                Sym(
+                    sym_name,
+                    sym_value_int,
+                    sym_type if sym_type else None,
+                    sym_size_int,
+                    sym_vrom_int,
+                )
+            )
+
+            if sym_name == target_sym_name:
+                target_vma = sym_value_int
+
+    if target_vma is None:
+        parser.error(f"No symbol '{target_sym_name}'")
+    else:
+        print(f"{target_sym_name} = 0x{target_vma:08X}")
+
+    filesections = list[FileSection]()
+
+    for file, syms_by_section in syms_by_section_by_file.items():
+        for section, syms in syms_by_section.items():
+            syms.sort(key=lambda sym: sym.value)
+            vma_start = syms[0].value
+            filesections.append(FileSection(file, section, syms, vma_start))
+
+    filesections.sort(key=lambda fs: fs.vma_start)
+
+    def get_first_print_sym():
+        prev_syms = list[Sym]()
+        for fs in filesections:
+            for sym in fs.syms:
+                if not args.labels and sym.type in LABELS_TYPES:
+                    continue
+                if target_vma < sym.value:
+                    return prev_syms[0]
+                prev_syms.append(sym)
+                while (
+                    len(prev_syms) - 1 > args.around
+                    and prev_syms[0].value < target_vma - args.range
+                ):
+                    prev_syms.pop(0)
+
+    first_print_sym = get_first_print_sym()
+
+    def get_last_print_sym():
+        min_skip_count = args.around
+        for fs in filesections:
+            for sym in fs.syms:
+                if not args.labels and sym.type in LABELS_TYPES:
+                    continue
+                if target_vma <= sym.value:
+                    min_skip_count -= 1
+                    if min_skip_count < 0 and sym.value >= args.range + target_vma:
+                        return sym
+
+    last_print_sym = get_last_print_sym()
+
+    is_near_target = False
+
+    indent = " " * 4
+
+    for i_fs, fs in enumerate(filesections):
+        fs_printed = False
+        is_first_fs_sym = True
+        fs_printed_end_ellipsis = False
+        for sym in fs.syms:
+            if not args.labels and sym.type in LABELS_TYPES:
+                continue
+
+            if sym == first_print_sym:
+                is_near_target = True
+
+            print_sym = is_near_target or (
+                args.file
+                and (
+                    fs.vma_start
+                    <= target_vma
+                    < (
+                        filesections[i_fs + 1].vma_start
+                        if i_fs + 1 < len(filesections)
+                        else 0x1_0000_0000
+                    )
+                )
```

Comment:
might just be the formatting, but personally I'd rather read
`fs.vma_start <= target_vma and (i_fs + 1 < len(filesections) and target_vma < filesections[i_fs + 1].vma_start)` or something. Or factor out into a variable `vma_end`?